packages/eslint-plugin/rules/jsx-a11y-anchor-has-content.js

```diff
@@ -0,0 +1,173 @@
+/**
+ *
+ * @file Wraps eslint-plugin-jsx-a11y/anchor-has-content to allow usage with createInterpolateElement.
+ *
+ */
+
+/**
+ * External dependencies
+ *
+ * eslint-plugin-jsx-a11y/anchor-has-content: to create a wrapper around it, we need to use it
+ */
+const anchorRule = require( 'eslint-plugin-jsx-a11y' ).rules[
+	'anchor-has-content'
+];
+
+/**
+ * Internal dependencies
+ */
+const {
+	getTextContentFromNode,
+	getTranslateFunctionName,
+	getTranslateFunctionArgs,
+} = require( '../utils' );
+
+const meta = {
+	type: 'problem',
+	docs: {
+		url: 'https://github.com/WordPress/gutenberg/tree/HEAD/packages/eslint-plugin/docs/rules/jsx-a11y-anchor-has-content.md',
+		description:
+			'Checks that anchors are not empty and have an href, relies on jsx-a11y/anchor-has-content under the hood. Supports usage of createInterpolateElement.',
+	},
+	fixable: 'code',
+	messages: {
+		anchorIsEmpty:
+			'Anchors must have content and the content must be accessible by a screen reader. Check the first parameter to createInterpolateElement.',
+		invalidMarkup:
+			'The first parameter to createInterpolateElement does not contain valid markup for an anchor.',
+	},
+	schema: [
+		{
+			type: 'object',
+			properties: {
+				components: {
+					type: 'array',
+					items: {
+						type: 'string',
+					},
+					uniqueItems: true,
+					additionalItems: false,
+				},
+			},
+		},
+	],
+};
+
+/**
+ *
+ * Verify that the content of anchors is not empty, and that we find an anchor in the markup. The anchors' content
+ * can be from an <a> or other tags that are configured in the eslint 'component' rules (e.g. <MyAnchor>)
+ *
+ * @param {string} markup   markup used in createInterpolateElement
+ * @param {string} tagNames names that we recognize as anchors, as configured in the eslint rules
+ * @throws {Error}              an error if the markup does not contain a valid anchor or is empty
```

Comment:
if I align this correctly, ESLint complains, *but* it is easy enough to solve it by configuring the jsdoc.js linter rules so that 'throws' is also counted as something that should be aligned. Happy to open a quick PR for that while this one is getting wrapped up, although the downside is that it'd cause a lot of new lint issues (I could fix them, I'm just not sure if we'd want to do it so I want to check first) 

```

		'jsdoc/check-line-alignment': [
			'error',
			'always',
			{
				tags: [ 'param', 'arg', 'argument', 'property', 'prop', 'throws' ],
				preserveMainDescriptionPostDelimiter: true,
			},
		],
```

Reply:
note: actually it wouldn't cause many  issues (I think there's just one other throw :) ) but it's still TBD if we want the change or not 

Changing the `@return` rule, on the other hand... would cause thousands of --fix changes.